wgpu-hal/src/metal/conv.rs

```diff
@@ -27,6 +27,11 @@ pub fn map_texture_usage(
         format.is_combined_depth_stencil_format(),
     );
 
+    mtl_usage.set(
+        metal::MTLTextureUsage::ShaderAtomic,
+        format == wgt::TextureFormat::R64Uint,
```

Comment:
There should be a better check here

Reply:
Im not sure what the best way to go about this would be. hal::TextureUses comes from wgt::TextureUsages (interesting that these are inconsistently named) which comes from webgpu spec apparently, which makes no accommodation for specifying atomic image usage.

Reply:
I think it's cleanest and best to just go ahead and add appropriate bits to `wgpu_types::TextureUsages` and then `wgpu_hal::TextureUses`. As long as they are clearly documented as extensions, with a reference to which `wgpu_types::Features` bit you need to turn on in order to use them, I think we're good.

Since these are the first extension bits we'd be adding to `wgpu_types::TextureUsages`, I'd suggest following the example of `wgpu_types::Features` which blocks off standard and extension portions:

```rust
        //
        // ---- Start numbering at 1 << 0 ----
        //
        // WebGPU features:
        //
...

        //
        // ---- Restart Numbering for Native Features ---
        //
        // Native Features:
        //
```

(I don't think that layout succeeds in making the section breaks stand out visually, but, whatever.)

It might also be good to start assigning bits after a gap. At the moment, `TextureUsages` only goes up to `1 << 4`, but it's reasonable to assume that the WebGPU standard will add more in the future. I mean, we don't promise any kind of binary interface stability, but it's still probably nice to avoid having our bits overlap with the standard's bits. So maybe `const SHADER_ATOMIC = 1 << 16;` or something.

(Although `wgpu_types::Features` does this too, it doesn't really matter, because the WebGPU spec never assigns bits to features, it only ever talks about them in terms of strings.)

Reply:
thanks, that makes sense. done